Client/Get.cs

```diff
@@ -21,50 +21,74 @@ public static int MultipleFiles(IEnumerable<string> files)
             return 0;
         }
 
-        public static int List(ref FtpClient client, Commands.List directory)
+        public static int List(ref FtpClient client, Commands.List directory, in Program.FilePath path, in string[] args)
         {
-            Console.WriteLine();
-            string path = "./";
-            try
+            if (args.Contains("-r"))
             {
-                if (directory.Local != null)
+                try
                 {
-                    path = Path.GetFullPath(directory.Local);
+                    Console.WriteLine(path.Remote);
+                    FtpListItem[] items = client.GetListing(path.Remote);
+
+                    foreach (FtpListItem item in items)
+                        Console.WriteLine(item.Name);
                 }
-                else if (directory.Remote != null)
+                catch (Exception ex)
                 {
-                    throw new InvalidOperationException("listing remote files (ls -r) not implemented");         // to be implemented!
+                    if (ex.InnerException != null)
+                        Console.WriteLine(ex.InnerException.Message);
+                    else
+                        Console.WriteLine(ex.Message);
+                    return 0;
                 }
+                return 1;
+            }
 
-                DirectoryInfo dir = new DirectoryInfo(path);
-                DirectoryInfo[] sub_directories = dir.GetDirectories();
-                FileInfo[] files = dir.GetFiles();
-
-                // List sub-directories
-                foreach (DirectoryInfo i in sub_directories)
+            else if (args.Contains("-l"))//i'll let Peter configure this part for his section
+            {
+                Console.WriteLine();
+                string filePath = "./";
+                try
                 {
-                    Console.WriteLine("{0}/", i.Name);
-                }
+                    if (directory.Local != null)
+                    {
+                        filePath = Path.GetFullPath(directory.Local);
+                    }
+                    else if (directory.Remote != null)
+                    {
+                        throw new InvalidOperationException("listing remote files (ls -r) not implemented");         // to be implemented!
+                    }
```

Comment:
Can this clause be removed now?

Reply:
Probably. That local listing was someone else's task, so I did not want to mess with their function.